src/object.rs

```diff
@@ -245,9 +251,9 @@ impl Ref {
 
 impl Primitive for Ref {
     #[inline]
-    fn write(self, buf: &mut Vec<u8>) {
+    fn write(self, buf: &mut Buf) {
         buf.push_int(self.0.get());
-        buf.extend(b" 0 R");
+        buf.extend_slice(b" 0 R");
```

Comment:
Would be nice if `Buf` would impl `Extend<u8>`. Then we could revert all of the `extend` -> `extend_slice` changes.